roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
-MB should work with Kusama validator specifications. For scaling up even more,
-runtime upgrades and message passing should be done off chain at some point.
-
-Flood protection considerations: For making DoS attacks slightly harder on this
-subsystem, nodes will only respond to large statement requests, when they
-previously notified that peer via gossip about that statement. So, it is not
-possible to DoS nodes at scale, by requesting candidate data over and over
-again.
+To address the concern of dealing with large numbers of spam candidates or
+statements, the overall design approach is to combine a focused "clustering"
+protocol for legitimate fresh candidates with a broad-distribution "grid"
+protocol to quickly get backed candidates into the hands of many validators.
+Validators do not eagerly send each other heavy `CommittedCandidateReceipt`,
+but instead request these lazily through request/response protocols.
+
+A high-level description of the protocol follows:
+
+### Messages
+
+Nodes can send each other a few kinds of messages: `Statement`,
+`BackedCandidateManifest`, `BackedCandidateAcknowledgement`.
+
+-   `Statement` messages contain only a signed compact statement, without full
+    candidate info.
+-   `BackedCandidateManifest` messages advertise a description of a backed
+    candidate and stored statements.
+-   `BackedCandidateAcknowledgement` messages acknowledge that a backed candidate
+    is fully known.
+
+### Request/response protocol
+
+Nodes can request the full `CommittedCandidateReceipt` and
+`PersistedValidationData`, along with statements, over a request/response
+protocol. This is the `AttestedCandidateRequest`; the response is
+`AttestedCandidateResponse`.
+
+### Importability and the Hypothetical Frontier
+
+-   The ****prospective parachains**** subsystem maintains prospective "fragment
+    trees" which can be used to determine whether a particular parachain candidate
+    could possibly be included in the future. Candidates which either are within a
+    fragment tree or <span class="underline">would be</span> part of a fragment tree if accepted are said to be
+    in the "hypothetical frontier".
+-   The ****statement-distribution**** subsystem keeps track of all candidates, and
+    updates its knowledge of the hypothetical frontier based on events such as new
+    relay parents, new confirmed candidates, and newly backed candidates.
+-   We only consider statements as "importable" when the corresponding candidate
+    is part of the hypothetical frontier, and only send "importable" statements to
+    the backing subsystem itself.
+
+### Cluster Mode
+
+-   Validator nodes are partitioned into groups (with some exceptions), and
+    validators within a group at a relay-parent can send each other `Statement`
+    messages for any candidates within that group and based on that relay-parent.
+    This is referred to as the "cluster" mode.
```

Comment:
Are these groups different from backing groups? I suppose it's important to mention it.

Reply:
It's the same as backing groups - this is lifted from the PR description of #5999 but I agree, worth noting that these are our same old backing groups used everywhere.